### PR TITLE
Added French, Dutch, German, Spanish, and Portuguese translation.

### PR DESCRIPTION
**Pull request by Davidthecode**
I added French, Dutch, German, Spanish, and Portuguese translation. so it is available in all 5 languages. The overall structure does not change. the only change i made to the UI was to add a select dropdown in the homepage where a user can select any of the 5 languages to continue with. i implemented it with next-translate package.

### CODE REVIEW

**Comment by Davidthecode** [comment]
This pr helps with the issue #3

**Comment by kenzivoy** [comment]
looks great

File: src/components/Form/Form.jsx [hunk 1]
```diff
@@ -93,40 +97,40 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
...
+              <label htmlFor="businessName" className={styles.label}>{t('name')}</label>
               <input 
                 className={styles.input__default}
                 type="text" 
                 name="businessName" 
                 id="businessName" 
                 placeholder="Business Name"
                 onChange={handleChange}
-                value={prefill.businessName || ''}  
+                value={`${t('business_name_placeholder')}` || ''}
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]: Seems the translation is assigned to the value instead of the placeholder.

File: src/components/Form/Form.jsx [hunk 2]
```diff
@@ -46,7 +48,9 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
     })
     return sum;
   }
-
+  
+  //useTranslation
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]: Did you intend to leave this comment?

File: src/components/Home/HomePage.jsx [hunk 3]
```diff
@@ -1,25 +1,31 @@
 import styles from './homePage.module.scss';
 import Link from 'next/link';
-
+import useTranslation from 'next-translate/useTranslation'
```
Comment (LaniJ) [inline src/components/Home/HomePage.jsx]:
```suggestion
import useTranslation from 'next-translate/useTranslation';
```

File: src/components/Home/HomePage.jsx [hunk 4]
```diff
@@ -1,25 +1,31 @@
...
+import LanguageChange from '../Language/languageChange';
 
 const HomePage = () => {
-  return ( 
+  const { t, lang } = useTranslation('common')
```
Comment (LaniJ) [inline src/components/Home/HomePage.jsx]:
```suggestion
  const { t, lang } = useTranslation('common');
```

File: src/components/InvoiceTemplate/InvoiceTemplate.jsx [hunk 5]
```diff
@@ -5,16 +5,18 @@ import invTemp2 from '../../assets/images/invTemp2.png';
 import invTemp3 from '../../assets/images/invTemp3.png';
 import invTemp4 from '../../assets/images/invTemp4.png';
 
+import useTranslation from "next-translate/useTranslation";
 
-const InvoiceTemplate = ({ template, changeTemplate }) => {
 
+const InvoiceTemplate = ({ template, changeTemplate }) => {
+  const {t} = useTranslation('common')
```
Comment (LaniJ) [inline src/components/InvoiceTemplate/InvoiceTemplate.jsx]:
```suggestion
  const {t} = useTranslation('common');
```

File: src/components/Language/languageChange.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import useTranslation from 'next-translate/useTranslation'
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+
+const LanguageChange = () => {
+    const { t,   lang } = useTranslation('common')
```
Comment (LaniJ) [inline src/components/Language/languageChange.jsx]: extra space between destructured values. Please Add semi-colons to the file in general.
Also could we properly format the code. Maybe Prettier?

File: src/components/Language/languageChange.jsx
Comment (LaniJ) [inline src/components/Language/languageChange.jsx:1]: Let's use Pascal case for file naming please.

File: src/components/Preview/Templates/Template1.jsx [hunk 7]
```diff
@@ -176,7 +176,7 @@ const Template1 = ({totalAmount, rows, logo, notes, currencySymbol, formName, bu
             <View style={styles.invoice_header}>
               <Text style={[styles.table_header, {width: '40%'}]}>DESCRIPTION</Text>
               <Text style={[styles.table_header, {width: '20%', textAlign: 'center'}]}>PRICE</Text>
-              <Text style={[styles.table_header, {width: '20%', textAlign: 'center'}]}>QTY</Text>
+              <Text style={[styles.table_header, {width: '20%', textAlign: 'center'}]}>QUANTITY</Text>
```
Comment (LaniJ) [inline src/components/Preview/Templates/Template1.jsx]: Noticed this text change, kindly revert

File: src/components/Table/Table.jsx [hunk 8]
```diff
@@ -14,6 +16,10 @@ const Table = ({ rows, currencySymbol, onModifyTable, onAddInvoiceRow, onRemoveI
       return amount;
   }
   
+
+  // translate
```
Comment (LaniJ) [inline src/components/Table/Table.jsx]: Did you intend to leave this comment?

File: src/components/Table/Table.jsx [hunk 9]
```diff
@@ -14,6 +16,10 @@ const Table = ({ rows, currencySymbol, onModifyTable, onAddInvoiceRow, onRemoveI
...
+  const {t} = useTranslation('common')
```
Comment (LaniJ) [inline src/components/Table/Table.jsx]:
```suggestion
  const {t} = useTranslation('common');
```

File: src/components/Preview/Preview.jsx [hunk 10]
```diff
@@ -6,7 +6,7 @@ import Template2 from "./Templates/Template2";
 import Template3 from "./Templates/Template3";
 import Template4 from "./Templates/Template4";
 
-const PDF = ({ template, rows, currencySymbol, formName, logo, logoUpdated, email, businessName, address, city, zipcode, phone, owner, clientName, clientEmail, clientAddress, clientCity, clientZipcode, clientPhone, date, InvoiceNo, website, notes, totalAmount }) => {
+const PDF = ({ template, rows, currencySymbol, formName, logo, logoUpdated, email, businessName, address, city, zipcode, phone, owner, clientName, clientEmail, clientAddress, clientCity, clientZipcode, clientPhone, date, InvoiceNo, website, notes, totalAmount, t, lang }) => {
```
Comment (LaniJ) [inline src/components/Preview/Preview.jsx]: Seems we aren't using these values. could we revert the changes?

File: src/components/Form/Form.jsx [hunk 11]
```diff
@@ -46,7 +48,9 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
...
+  const {t} = useTranslation('common')
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]:
```suggestion
  const {t} = useTranslation('common');
```

File: src/components/Form/Form.jsx [hunk 12]
```diff
@@ -150,72 +154,72 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
...
+              <label htmlFor="clientName" className={styles.label}>{t('name')}</label>
               <input 
                 className={styles.input__default}
                 type="text" 
                 name="clientName" 
                 id="clientName" 
                 placeholder="Business Name"
                 onChange={handleChange}
-                value={prefill.clientName || ''} 
+                value={`${t('business_name_placeholder')}` || ''}
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]: translation value is assigned to the value attribute instead of the placeholder so users can't type in the input field since it's hard coded

File: src/components/Form/Form.jsx [hunk 13]
```diff
@@ -150,72 +154,72 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
...
+              <label htmlFor="clientEmail" className={styles.label}>{t('email')}</label>
                 <input
                   className={styles.input__default}
                   type="email" 
                   name="clientEmail" 
                   id="clientEmail" 
                   placeholder="name@clientemail.com"
                   onChange={handleChange}
-                  value={prefill.clientEmail || ''} 
+                  value={`${t('client_email_placeholder')}` || ''}
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]: translation value is assigned to the value attribute instead of the placeholder so users can't type in the input field since it's hard coded

File: src/components/Form/Form.jsx [hunk 14]
```diff
@@ -239,21 +243,21 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
                 type="text" 
                 name="clientZipcode" 
                 id="clientZipcode" 
-                placeholder="postal code"
+                placeholder={t('postal_code_placeholder')}
                 onChange={handleChange}
-                value={prefill.clientZipcode || ''} 
+                value={`${t('postal_code_placeholder')}` || ''}
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]: here, the translation is assigned to both value attribute and the placeholder.

File: src/components/Form/Form.jsx [hunk 15]
```diff
@@ -239,21 +243,21 @@ const Form = ({ logo, updateLogo, logoUpdated, prefill, currencySymbol, rows, on
...
+              <label htmlFor="clientPhone" className={styles.label}>{t('phone')}</label>
               <input
                 className={styles.input__default}
                 type="tel" 
                 name="clientPhone" 
                 id="clientPhone" 
                 placeholder="(123) 456 789"
                 onChange={handleChange}
-                value={prefill.clientPhone || ''} 
+                value={`${t('phone_placeholder')}` || ''}
```
Comment (LaniJ) [inline src/components/Form/Form.jsx]: the translation value is assigned to the value attribute instead of the placeholder. input field value is hard coded

**Review by LaniJ — COMMENTED** [review]
Hi @Davidthecode 
Thanks so much for taking out time to do this awesome work! 😊
It's much appreciated.
I've left feedback for some minor changes.

**Comment by LaniJ** [comment]
For the language dropdown, could it be moved to the top right corner?
Then, we wouldn't need the "language" label, it would just be a dropdown with the language abbreviations. 
E.g - EN for English 

Screenshot 2023-08-06 at 11 22 22

**Comment by RoneoOrg** [comment]
Glad to see that translations are coming.
Feel free to ping me if the French translation requires a review

File: src/components/Form/Form.jsx (hunk 2 above)
Comment (Davidthecode) [reply src/components/Form/Form.jsx]: The comment has been removed

**Comment by Davidthecode** [comment]
Hi @LaniJ, i have made all the changes

**Comment by LaniJ** [comment]
Nicely done.
Only thing i notice is the the dropdown doesn't persist the displayed selection when the page is changed. The selected language is still correct but the dropdown value shows 'en' by default.
Could you pick the value from local storage instead? Everything else looks good to me. 😊

**Comment by Davidthecode** [comment]
oh my bad, i have fixed that @LaniJ

**Comment by Davidthecode** [comment]
Hi @LaniJ, I'm waiting on your feedback

**Comment by LaniJ** [comment]
Hiya @Davidthecode  Looks good. thanks for contributing! 😊

**Comment by Davidthecode** [comment]
🚀🚀
